service/controller/clusterapi/v16/key/cluster.go

```diff
@@ -1,14 +1,65 @@
 package key
 
 import (
+	"github.com/giantswarm/cluster-operator/pkg/label"
 	"github.com/giantswarm/microerror"
 	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
 )
 
+func AWSClusterConfigName(cluster v1alpha1.Cluster) string {
+	return clusterProviderStatus(cluster).Cluster.ID
+}
+
+func ClusterAvailabilityZones(cluster v1alpha1.Cluster, machineDeployments []v1alpha1.MachineDeployment) []string {
+	azMap := make(map[string]struct{})
+
+	azMap[ClusterMasterAZ(cluster)] = struct{}{}
+
+	// TODO: Extract AZs from MachineDeployments
+
+	azs := make([]string, 0, len(azMap))
+	for az := range azMap {
+		azs = append(azs, az)
+	}
+	return azs
+}
+
+func ClusterCredentialSecretName(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.CredentialSecret.Name
+}
+
+func ClusterCredentialSecretNamespace(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.CredentialSecret.Namespace
+}
+
+func ClusterDNSZone(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Cluster.DNS.Domain
+}
+
 func ClusterID(cluster v1alpha1.Cluster) string {
 	return clusterProviderStatus(cluster).Cluster.ID
 }
 
+func ClusterMasterAZ(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.Master.AvailabilityZone
+}
+
+func ClusterMasterInstanceType(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).Provider.Master.InstanceType
+}
+
+func ClusterName(cluster v1alpha1.Cluster) string {
+	return clusterProviderSpec(cluster).ClusterName
+}
+
+func ClusterReleaseVersion(cluster v1alpha1.Cluster) string {
+	relVer, ok := cluster.Labels[label.ReleaseKey]
+	if !ok {
+		relVer = "unknown"
```

Comment:
This is not ideal, but I don't want to panic on invalid input either. That would cause weird `CrashLoopBackOff` and prevent normal operation while the end-user might be completely unaware of the effects of missing label on the CR.

`unknown` won't proceed in correctly when fetching version bundles and leads to error in cluster config mapping and therefore provides reasonable outcome while not blocking other clusters.

Reply:
Ok. I'll change this to be a panic.